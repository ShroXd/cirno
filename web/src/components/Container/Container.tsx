export const Container = ({
  children,
  className,
}: {
  children: React.ReactNode
  className?: string
}) => <div className={`px-6 pb-6 ${className}`}>{children}</div>
