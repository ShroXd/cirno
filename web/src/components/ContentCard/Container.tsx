import { ReactNode } from 'react'

import { Card } from '@material-tailwind/react'

interface ContentCardContainerProps {
  children: ReactNode
  className?: string
}

export const ContentCardContainer = ({
  children,
  className,
}: ContentCardContainerProps) => (
  <Card
    className={`flex select-none flex-col overflow-hidden rounded-xl bg-white shadow-lg ${className || ''}`}
  >
    {children}
  </Card>
)
