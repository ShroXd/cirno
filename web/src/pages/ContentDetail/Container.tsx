import { motion } from 'motion/react'

import { cn } from '~/lib/utils'

interface FadeTransitionContainerProps {
  className?: string
  children: React.ReactNode
}

export const FadeTransitionContainer = ({
  children,
  className,
}: FadeTransitionContainerProps) => (
  <motion.div
    initial={{ opacity: 0 }}
    animate={{ opacity: 1 }}
    exit={{ opacity: 0 }}
    transition={{ duration: 0.7, ease: 'easeInOut' }}
    className={cn(
      'relative mb-8 aspect-[21/9] w-full overflow-hidden rounded-t-xl',
      className
    )}
  >
    {children}
  </motion.div>
)
