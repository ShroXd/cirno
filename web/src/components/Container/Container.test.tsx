import { render } from '@testing-library/react'
import { describe, expect, it } from 'vitest'

import { Container } from './Container'

describe('Container', () => {
  it('should render correctly with children', () => {
    const { container } = render(<Container>Child</Container>)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render correctly with custom className', () => {
    const { container } = render(<Container className='mt-4'>Child</Container>)

    expect(container).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
