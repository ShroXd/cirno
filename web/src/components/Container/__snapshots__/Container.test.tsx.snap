// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > should render correctly with children 1`] = `
<div
  class="px-6 pb-6 undefined"
>
  Child
</div>
`;

exports[`Container > should render correctly with custom className 1`] = `
<div
  class="px-6 pb-6 mt-4"
>
  Child
</div>
`;
